src/btree/bt_page.c

```diff
@@ -242,20 +261,37 @@ __wt_page_inmem_prepare(WT_SESSION_IMPL *session, WT_REF *ref)
             /* For each record, create an update to resolve the prepare. */
             for (; rle > 0; --rle, ++recno) {
                 /* Create an update to resolve the prepare. */
-                WT_ERR(__page_inmem_prepare_update(session, value, &unpack, &upd, &size));
+                WT_ERR(__page_inmem_prepare_update_col(
+                  session, ref, &cbt, recno, value, &unpack, &upd, &size));
                 total_size += size;
-
-                /* Search the page and apply the modification. */
-                WT_ERR(__wt_col_search(&cbt, recno, ref, true, NULL));
-#ifdef HAVE_DIAGNOSTIC
-                WT_ERR(__wt_col_modify(&cbt, recno, NULL, upd, WT_UPDATE_INVALID, true, true));
-#else
-                WT_ERR(__wt_col_modify(&cbt, recno, NULL, upd, WT_UPDATE_INVALID, true));
-#endif
                 upd = NULL;
             }
         }
+    } else if (page->type == WT_PAGE_COL_FIX) {
+        WT_ASSERT(session, WT_COL_FIX_TWS_SET(page));
```

Comment:
I've always felt it's better to assert than crash, because then you can tell what happened even if you didn't have a debugger attached. Also in this case I think it got put there to reassure anyone wondering if this code _can_ crash on the next line. So I'd probably prefer to leave it...

Reply:
I won't fight about it.